### PR TITLE
ci: check broken link

### PR DESCRIPTION
**Pull request by xuruidong**
### Description

1. add ci configuration of check-broken-link
2. fix broken links

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
@kayx23 please help to check this

**Comment by monkeyDluffy6017** [comment]
@guoqqqi please help to check this

**Comment by xuruidong** [comment]
Some more information about this PR:
1. I fixed all the broken relative path of markdown files
2. I fixed most of the broken external links, some links I don't konw how to fix
3. The ci configuration of "link checker" only check relative path of markdown files, and the external link checking is disabled (the default behave)

**Review by lingsamuel — APPROVED** [review]
great work, thanks!

**Comment by monkeyDluffy6017** [comment]
Excellent job!!!

**Review by guoqqqi — APPROVED** [review]
LGTM

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+jobs:
+  check-links:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get script
+        run: |
+          wget https://raw.githubusercontent.com/xuruidong/markdown-link-checker/main/link_checker.py
```
Comment (kayx23) [inline .github/workflows/link-check.yml:39]: @monkeyDluffy6017 note that this is a personal repo.

**Comment by kayx23** [comment]
> 3\. The ci configuration of "link checker" only check relative path of markdown files, and the external link checking is disabled (the default behave)

@xuruidong I see you fixing external links in this PR as well. Did you manually check then?

Thanks for your contribution btw!

**Pull request merged**